app/models/katello/concerns/http_proxy_extensions.rb

```diff
@@ -8,10 +8,17 @@ module HttpProxyExtensions
       end
 
       def update_default_proxy_setting
-        setting = Setting.where(name: "content_default_http_proxy").first
+        changes = self.previous_changes
+        if changes.key?(:name)
+          previous_name = changes[:name].first
+          setting = Setting.
+            where(name: "content_default_http_proxy").
+            where(value: previous_name.to_yaml).
```

Comment:
why did you have to call .to_yaml?

Reply:
If you attempt to query based on the proxy name, the where clause will match nothing. The setting values are converted to yaml during the assignment of values.
See 
https://github.com/theforeman/foreman/blob/a31356d3f35ed8f2da7eb1501c8cf9a2551855e3/app/models/setting.rb#L116